libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshStub.kt

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.navigation.core.routerefresh
+
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.NavigationRouterRefreshCallback
+import com.mapbox.navigation.base.route.RouterFactory.buildNavigationRouterRefreshError
+import com.mapbox.navigation.core.directions.session.RouteRefresh
+
+@OptIn(ExperimentalMapboxNavigationAPI::class)
+class RouteRefreshStub : RouteRefresh {
+
+    private var requestId = 0L
+    private val handlers = mutableMapOf<String, RouteRefreshHandler>()
+
+    override fun requestRouteRefresh(
+        route: NavigationRoute,
+        legIndex: Int,
+        callback: NavigationRouterRefreshCallback
+    ): Long {
+        val currentRequestId = requestId++
+        val handler = handlers[route.id]
+        if (handler != null) {
+            handler(route, legIndex, callback)
+        } else {
+            callback.onFailure(buildNavigationRouterRefreshError("handle isn't configured yet"))
+        }
+
+        return currentRequestId
+    }
+
+    override fun cancelRouteRefreshRequest(requestId: Long) {
+    }
+
+    /***
+     * Tne next route refresh requests will return the actual routes
+     */
+    fun setRefreshedRoute(refreshedRoute: NavigationRoute) {
+        handlers[refreshedRoute.id] = { _, _, callback ->
+            // TODO: refresh legs only from the passed index
+            callback.onRefreshReady(refreshedRoute)
+        }
+    }
+
+    /***
+     * The next route refresh requests for the given route route will fail
+     */
+    fun failRouteRefresh(navigationRouteId: String) {
+        handlers[navigationRouteId] = { _, _, callback ->
```

Comment:
stub contains business logic, that also requires tests

Reply:
That's what I'm talking about. It's too complex. Although I once covered unit tests with unit tests. :)

Reply:
I suppose we can have stubs and mocks at the same project depending on purposes. It only requires keeping stubs not over complicated and as you mentioned writing tests on tests where it needs 👿

Reply:
added tests. @RingerJK , @dzinad , is it easier to understand what the stub does now?